system-tests/e2e-suite/src/test/scala/com/hotels/styx/proxy/BadResponseFromOriginSpec.scala

```diff
@@ -95,16 +96,7 @@ class BadResponseFromOriginSpec extends FunSpec
+          HttpHeader("Accept-Language", Random.alphanumeric.take(1000).mkString),
```

Comment:
Is this test supposed to fail due to too long headers? I have completely forgotten.

We should emphasise its intention by explicitly configuring the `maxHeaderSize` in the test fixture settings. Otherwise all future maintainers will be always guessing the test intention. 

Like so:

```
  override protected def beforeAll() = {
    super.beforeAll()
    styxServer.setBackends(
      "/badResponseFromOriginSpec/" -> HttpBackend(
        "app-1", Origins(originOne),
        connectionPoolConfig = ConnectionPoolSettings(maxConnectionsPerHost = 1)

        // HERE: you'd have to add this:
        maxHeaderSize = 999
      )
    )
  }
```

But as we are migrating away from Scala tests, I would take this opportunity to move it over to the Kotlin FT suite.

Reply:
It was indeed failing due to long headers... We have another test for the code path in which we are using HostProxy instead of BackendServicesRouter , and we are also migrating away from the latter so I don't think porting the tests is too useful in this case.